src/cargo/ops/lockfile.rs

```diff
@@ -19,20 +20,25 @@ pub fn load_pkg_lockfile(pkg: &Package) -> CargoResult<Option<Resolve>> {
 
 pub fn load_lockfile(path: &Path, sid: &SourceId) -> CargoResult<Option<Resolve>> {
     // If there is no lockfile, return none.
-    let mut f = match File::open(path) {
-        Ok(f) => f,
+    let s = match load_string(path) {
```

Comment:
The semantics here are slightly different than before, and I'd prefer to leave them as before for now. An error when opening the file is interpreted as "file not found", aka an OK situation where `None` is returned, but an error when reading the file should be treated as fatal as it did indeed exist and some weird error happened.